src/contributor-guide/dod/tco-analysis.md

```diff
@@ -0,0 +1,105 @@
+|                                        | Method parameter default value changed                                                                                   | MAJOR |
+|                                        | Method parameter default value added                                                                                     | MINOR |
+|                                        |                                                                                                                          |       |
+| PHP Class (marked with @api)           | Method argument or return value format changed (Example: before - [1, 2, 3], after - ['items' => [1,2,3], 'count' => 3]) | MAJOR |
+|                                        | Trait removed                                                                                                            | MAJOR |
+| Message queue                          | Topic removed                                                                                                            | MAJOR |
+|                                        | Consumer removed                                                                                                         | MINOR |
+|                                        | New topic published                                                                                                      | MINOR |
+| Layout handles declared by modules     | New layout page handle                                                                                                   | MINOR |
+|                                        | New container/block added to handle                                                                                      | MINOR |
+|                                        | Removed layout handle                                                                                                    | MAJOR |
+| Static view files                      | JS/CSS/LESS file removed/moved                                                                                           | MAJOR |
+
+### Potential number of affected customisations
+
+In order to identify the volume of the impact the usages of the changed code should be detected in the available code/extensions/customisations.
+
+The analysis result should indicate the proportion of reviewed modules to the modules referencing (using) the changed code.
+
+### TCO Analysis result example
+
+The analysis result should indicate all the changes that have TCO impact (or simply state that there is "No TCO impact").
+
+For example, if the non-api "SomeClass::someMethod" was changed and the new required argument was added to the CLI command, the analysis result may look like:
+
+```text
+Changes have TCO impact.
+Change to SomeClass::someMethod is PATCH level but impacts 5 out of 10 reviewed projects.
```

Comment:
There should be some instructions for how to perform this analysis. Including details such as which projects could be reviewed, how many should be reviewed (if any). 

In general I found this example analysis to be confusing because this is the first place in this document where it is mentioned that other projects should be reviewed.

Reply:
@nathanjosiah the Adobe commerce projects available to the contributor should be reviewed. I will try to add more content to make this clear.

Reply:
@sivaschenko What does this mean to internal developers working on core. It's mostly just security and platform health but we are still making changes. Which projects should we review?

Reply:
@nathanjosiah I meant "Adobe Commerce project" is a web store that is based on the magento2 framework. Updated to `custom modules` to avoid such confusion